pkg/proxy/userspace/proxysocket.go

```diff
@@ -216,54 +230,51 @@ func (udp *udpProxySocket) ProxyLoop(service proxy.ServicePortName, myInfo *serv
 			glog.Errorf("ReadFrom failed, exiting ProxyLoop: %v", err)
 			break
 		}
-		// If this is a client we know already, reuse the connection and goroutine.
-		svrConn, err := udp.getBackendConn(myInfo.activeClients, cliAddr, proxier, service, myInfo.timeout)
-		if err != nil {
-			continue
-		}
-		// TODO: It would be nice to let the goroutine handle this write, but we don't
-		// really want to copy the buffer.  We could do a pool of buffers or something.
-		_, err = svrConn.Write(buffer[0:n])
-		if err != nil {
-			if !logTimeout(err) {
-				glog.Errorf("Write failed: %v", err)
-				// TODO: Maybe tear down the goroutine for this client/server pair?
-			}
-			continue
-		}
-		err = svrConn.SetDeadline(time.Now().Add(myInfo.timeout))
-		if err != nil {
-			glog.Errorf("SetDeadline failed: %v", err)
-			continue
-		}
+		udp.connectBackend(myInfo.activeClients, &buffer, n, cliAddr, proxier, service, myInfo.timeout)
 	}
 }
 
-func (udp *udpProxySocket) getBackendConn(activeClients *clientCache, cliAddr net.Addr, proxier *Proxier, service proxy.ServicePortName, timeout time.Duration) (net.Conn, error) {
+func (udp *udpProxySocket) connectBackend(activeClients *clientCache, buffer *[bufferSize]byte, n int, cliAddr net.Addr, proxier *Proxier, service proxy.ServicePortName, timeout time.Duration) {
```

Comment:
I'll use an anonymous function to wrap it and put inside the ProxyLoop function, since I think putting lots of code between lock/unlock pairs is harmful for readability.
If this is not the case or there is any other better approach please let me know.